src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -225,5 +225,28 @@ internal static bool IsValidUnscopedRefAttributeTarget(this MethodSymbol method)
                 !method.IsConstructor() &&
                 !method.IsInitOnly;
         }
+
+#nullable enable
+        internal static ImmutableArray<DeclarationScope> GetByValueParameterDeclaredScopes(this MethodSymbol? method)
+        {
+            if (method is null)
+                return default;
+
+            if (method.Parameters.All(p => getByValueParameterDeclaredScope(p) == DeclarationScope.Unscoped))
+                return default;
+
+            return method.Parameters.SelectAsArray(p => getByValueParameterDeclaredScope(p));
+
+            static DeclarationScope getByValueParameterDeclaredScope(ParameterSymbol parameter)
+            {
+                // For implicitly-typed lambda parameters, we only care about declaration scopes on by-value parameters.
```

Comment:
>For implicitly-typed lambda parameters, we only care about declaration scopes on by-value parameters. [](http://example.com/codeflow?start=19&length=101)

It is not obvious why is that.

Reply:
If a parameter is `ref` or `out` then the parameters must have been explicitly-typed.

Reply:
> If a parameter is ref or out then the parameters must have been explicitly-typed.

I am not sure I am following. We are looking at a method symbol. All parameters are explicitly typed for it.

Reply:
Still looking at other changes, I might get it at the end